### PR TITLE
chore: support mtls settings

### LINKED ISSUE #2211: Client proxy mTLS support
I need to send HTTP traffic though a proxy that requires mutual TLS authentication. There currently is no way to provide the required certificates to the client to do this.

### PR DESCRIPTION
**Pull request by rajcspsg**
closes https://github.com/zio/zio-http/issues/2211

### CODE REVIEW

**Comment by rajcspsg** [comment]
@adamgfraser still I need to write test. could you have quick glance whether the approach is correct.

**Review by jdegoes — APPROVED** [review]

**Comment by jdegoes** [comment]
@danestig Can you review and ensure this meets your needs?

**Comment by danestig** [comment]
Hello! Thanks for taking a stab at this @rajcspsg.

I have 2 concerns: 
1. How do I provide both client keytstore/key/certificate and a truststore/ca-certificate? Looking at the `ClientSSLConfig` they look mutually exclusive to me. 
2. Since I could not figure out 1 I have not tried this yet. Will mTLS be used to connect to both the proxy and the server, or only one of them? #2211 Is specifically about authenticating with mTLS with the proxy, I expected that to be configured in the proxy setting.

**Comment by rajcspsg** [comment]
@danestig Sorry I missed your comment. I did mistake I updating the ClientSSLConfig config instead of ServerSSLConfig. 
I going to configure only the proxy server verifies the client certificates, does that work for you?

**Comment by danestig** [comment]
@rajcspsg Yes, I believe that would solve my issue. Thank you.

File: zio-http/src/main/scala/zio/http/ClientSSLConfig.scala [hunk 1]
```diff
@@ -42,9 +52,22 @@ object ClientSSLConfig {
     )
   }
 
-  case object Default                                                                         extends ClientSSLConfig
-  final case class FromCertFile(certPath: String)                                             extends ClientSSLConfig
-  final case class FromCertResource(certPath: String)                                         extends ClientSSLConfig
-  final case class FromTrustStoreResource(trustStorePath: String, trustStorePassword: String) extends ClientSSLConfig
-  final case class FromTrustStoreFile(trustStorePath: String, trustStorePassword: String)     extends ClientSSLConfig
+  sealed trait ClientAuth
+  case object RequireClientAuth  extends ClientAuth
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/ClientSSLConfig.scala]: @rajcspsg Can you put these in the companion object of `ClientAuth`?

Comment (jdegoes) [reply zio-http/src/main/scala/zio/http/ClientSSLConfig.scala]: Also, once you put them there, they can have shorter names, e.g.:

```scala
object ClientAuth {
  case object Required extends ClientAuth
  case object None extends ClientAuth
  case object Optional extends ClientAuth
}
```

**Review by jdegoes — CHANGES_REQUESTED** [review]
Just one minor suggestion, then looks good to merge. Thankk you!

File: zio-http/src/main/scala/zio/http/ClientSSLConfig.scala (hunk 1 above)
Comment (rajcspsg) [reply zio-http/src/main/scala/zio/http/ClientSSLConfig.scala]: @jdegoes sure. I've to set this config on ServerSslContext not on ClientSSLContext

Comment (rajcspsg) [reply zio-http/src/main/scala/zio/http/ClientSSLConfig.scala]: I'll update my PR shortly

**Comment by rajcspsg** [comment]
@jdegoes @danestig @adamgfraser I updated the PR to add clientAuth config on SSLServerContext.
Please review again and let me know if you have any questions.

File: zio-http/src/main/scala/zio/http/SSLConfig.scala [hunk 2]
```diff
@@ -17,16 +17,32 @@
...
+)
 
 object SSLConfig {
 
   def apply(data: Data): SSLConfig =
-    new SSLConfig(HttpBehaviour.Redirect, data, Provider.JDK)
+    new SSLConfig(HttpBehaviour.Redirect, data, Provider.JDK, None)
+
+  def apply(data: Data, clientAuth: ClientAuth): SSLConfig =
+    new SSLConfig(HttpBehaviour.Redirect, data, Provider.JDK, Some(clientAuth))
```
Comment (rajcspsg) [inline zio-http/src/main/scala/zio/http/SSLConfig.scala:46]: @jdegoes @adamgfraser  I tried to combine both the apply (and other duplicate functions below)
into single function like 

```scala
def apply(data: Data, clientAuth: Option[ClientAuth] = None): SSLConfig =
    new SSLConfig(HttpBehaviour.Redirect, data, Provider.JDK, clientAuth)
```

But I'm getting below error -

 > /home/raj/Coding/scala/zio-http/zio-http/src/main/scala/zio/http/SSLConfig.scala:39:8: in object SSLConfig, multiple overloaded alternatives of method fromFile define default arguments.


I did workaround to make it compile. Not sure why the first option didn't work

**Review by jdegoes — APPROVED** [review]

**Pull request merged**